### PR TITLE
Added Train-A project

### PR DESCRIPTION
**Pull request by OlehDulebaEpam**
#### New final project: Train-A

Added .md files along with images to the project Train-A in `tasks`

#### 🤔 This is a ...

- [x] 🌟 New task
- [ ] 🌐 New module
- [ ] ⚙️ Update to an existing task
- [ ] 🔧 Update to an existing module
- [ ] 🔗 Update or addition of external resources or links
- [ ] 🐛 Fix in a task or related content
- [ ] 🛠 Fix in a module or related content
- [ ] ✏️ Fixed a typo or grammatical error
- [ ] 🔗 Fixed a broken link
- [ ] ❓ Other (specify: **\*\*\*\***\_\_\_\_**\*\*\*\***)

#### Description

- **Brief Overview:**
  New project composed to orchestrate train journeys as manager and consume as a user.
- **Implementation Approach:**
  Project requires Angular framework, libraries for components and styles.

#### Additional Information

- **Screenshots/Links:**
  
result-route

- [ ] Related Issues:

#### Checklist

- [x] ✅ I have performed a self-review of my own code.
- [x] 📝 I have commented my code, particularly in hard-to-understand areas.
- [x] 🔧 I have made corresponding changes to the documentation (if applicable).
- [x] 🚫 My changes generate no new warnings or errors.

### CODE REVIEW

**Comment by lyutails** [comment]
hey there o/

isn't final project lasts for three (3) weeks :?

image

image

and maybe worth adding info in readme about submitting task for cross-check :)

File: tasks/train-a/search.md [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+
+- Each result item element must have a "Route" button that is clearly visible.
+
+#### Acceptance Criteria 2: Modal window trigger (2)
+
+- When the "Route" button is clicked, a modal window must appear.
+
+#### Acceptance Criteria 3: Modal window content (5)
+
+- Title of modal contains text "Route" alogn with route id number.
```
Comment (lyutails) [inline tasks/train-a/search.md:225]: typo in 'alogn' should be along :)

File: tasks/train-a/header-fragment.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+- When a user is authenticated as a manager or administrator:
+  - The header must display "Home", "Profile", "My Orders", and "Admin" links.
+  - The "Home" link must redirect to the [home page](./search.md).
+  - The "Profile" link must redirect to the user's [profile page](./profile.md).
+  - The "My Orders" link must redirect to the user's [orders page](./order.md).
+  - The "Admin" link must redirect to the [admin page](./admin/readme.md).
+  - The "Admin" link must be accessible only to users with the manager or administrator role, enforced by a guard.
+
+#### Acceptance Criteria 4: Guard for qdmin page
```
Comment (lyutails) [inline tasks/train-a/header-fragment.md:59]: a little typo here :) 'qdmin' should be 'admin'

File: tasks/train-a/header-fragment.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+### Requirements
+
+#### Requirement 1: Header states
+
+- The header must support three different states based on the user's authentication and authorization level: Guest, General User, and Manager/Administrator.
+
+#### Requirement 2: Guest state
+
+- The header must display the following links for guests:
```
Comment (lyutails) [inline tasks/train-a/header-fragment.md:15]: a bit confusing ':' at the end of the line :)
maybe to add smth like `you can find requirements in the #### Acceptance Criteria 1: Guest state`

File: tasks/train-a/header-fragment.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+#### Requirement 3: General user state
+
+- The header must display the following links for authenticated general users:
```
Comment (lyutails) [inline tasks/train-a/header-fragment.md:19]: a bit confusing ':' at the end of the line :)
maybe to add smth like `#### Acceptance Criteria 2: General user state`

File: tasks/train-a/profile.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+## Description
+
+Page allows authenticated users to view and update their personal information. Users can change details such as their _name_, _email_, and _password_.
+
+### Core elements
+
+- **Name record:** Name of the user.
+- **Email record:** Registered user email.
+- **Eedit buttons:** Pencil buttons to allow user edit information.
```
Comment (lyutails) [inline tasks/train-a/profile.md:17]: possibly should be 'Edit' instead of 'Eedit' :)

File: tasks/train-a/search.md [hunk 6]
```diff
@@ -0,0 +1,374 @@
...
+
+### Core elements
+
+- **Search section**:
+  - **Form field _From_**: Where the trip starts
+  - **Form field _To_**: Where the trip ends
+  - **Form field _Date_**: From what date to search
+    - **Search**: Button to initiate the search process
+- **Result list item**:
+  - **Arrival date/time**: When train reach the target
```
Comment (lyutails) [inline tasks/train-a/search.md:35]: probably should be `reaches` here

File: tasks/train-a/search.md [hunk 7]
```diff
@@ -0,0 +1,374 @@
...
+- The form must include a submit button.
+- The submit button must be disabled until the form is valid (all fields are correctly filled).
+
+### Acceptance Criteria: Search section [(design 1)](#search-block-1) (20)
+
+#### Acceptance Criteria 1: Start city field autocomplete (6)
+
+> [!WARNING]
+> **Technical quirk**  
+> Developer should obtain coordinates of the location throught whatever external services their want while user fill city name in the form field.
```
Comment (lyutails) [inline tasks/train-a/search.md:67]: a typo :)
`throught` probably should be `through`

File: tasks/train-a/search.md [hunk 8]
```diff
@@ -0,0 +1,374 @@
...
+
+- The form must include a field labeled "From" for the start city.
+- When the user begins typing in the "From" field, an autocomplete dropdown must appear with city suggestions.
+- The user must be able to select a city from the autocomplete suggestions.
+
+#### Acceptance Criteria 2: End city field autocomplete (6)
+
+> [!WARNING]
+> **Technical quirk**  
+> Developer should obtain coordinates of the location throught whatever external services their want while user fill city name in the form field.
```
Comment (lyutails) [inline tasks/train-a/search.md:77]: probably same typos here
`throught` probably should be `through`
and maybe `their want` > `they want`
and probably `user fills`

**Review by Satlykov — APPROVED** [review]

**Pull request merged**